datafusion/optimizer/src/propagate_empty_relation.rs

```diff
@@ -142,13 +146,19 @@ impl OptimizerRule for PropagateEmptyRelation {
                             schema: join.schema.clone(),
                         }),
                     )),
+                    JoinType::LeftAnti if right_empty => {
```

Comment:
I double checked that an ANTI join (e.g. a `NOT IN`) is true when there are no tuples (aka the right side is emprt)

```sql
postgres=# select * from foo;
  x
-----
   1
   2
 NaN
(3 rows)

postgres=# select * from bar;
 y
---
(0 rows)

postgres=# select x NOT IN (SELECT y from bar) from foo;
 ?column?
----------
 t
 t
 t
(3 rows)
```

Thus I think this is a valid rewrite

Reply:
Nice, thanks for the double check!